File: src/app/model/book-loan.ts
import { Book } from "./book";

export interface BookLoan {
    loan_id: string;
    book_id: string;
    member_id: string;
    date_borrowed: Date;
    due_date: Date;
    date_returned: Date | null;
    status: string;
    fine_amount: number;
    book: Book;
  }
  

File: LibraryManagement.Core/Entities/BookLoan.cs
using System;
using System.Collections.Generic;

namespace LibraryManagement.Core.Entities;

public partial class BookLoan
{
    public string LoanId { get; set; } = null!;

    public string BookId { get; set; } = null!;

    public string MemberId { get; set; } = null!;

    public DateOnly DateBorrowed { get; set; }

    public DateOnly DueDate { get; set; }

    public DateOnly? DateReturned { get; set; }

    public string? Status { get; set; }

    public decimal? FineAmount { get; set; }

    public virtual Book Book { get; set; } = null!;

    public virtual ICollection<Fine> Fines { get; set; } = new List<Fine>();

    public virtual Member Member { get; set; } = null!;
}
